integreat_cms/cms/models/abstract_tree_node.py

```diff
@@ -282,7 +287,19 @@ def move(self, target, pos=None):
         # Moving a node can modify all other nodes via raw sql queries (which are not recognized by cachalot),
         # so we have to invalidate the whole model manually.
         invalidate_model(self.__class__)
-        super().move(target=target, pos=pos)
+        try:
+            with db_mutex("page-tree"):
```

Comment:
```suggestion
            with db_mutex(self.__class__.__name__):
```
Ah, just noticed this is the shared model between language tree and page tree, so technically we can use two locks for those two models :sweat_smile: 

Also, I thought maybe it would make sense to put the constraint check in the lines 270-286 into the lock-protected area as well? Otherwise the condition might not be true anymore once the lock is acquired.

Reply:
Ah yes, I missed that after moving the code from the Page class to the abstract class.

Reply:
What do you think about including the constraint check in the mutex area as well?

Reply:
You mean the database constraints for tree consistency? I'm not yet sure how that would work. The constraints just make some (bad) queries fail, don't they?

Reply:
I mean those in the lines above:

https://github.com/digitalfabrik/integreat-cms/blob/97fe210503c4233566303fa6c26cc192db4f064e/integreat_cms/cms/models/abstract_tree_node.py#L269-L286

I can imagine a rare edge case where this condition is `False` before acquiring the lock, but then in the mean time until the view waits for the lock, the condition becomes `True` which should cause the operation to fail in theory?

Reply:
If we move the last cache flush into the mutex protected block as well, couldn't we omit the first cache flush?

Reply:
Not sure, I guess there will still be a lot of cases where there is either a non-move operation before or after the move, which requires the flushing on both ends, right?
Only in the edge cases that two moves are directly consecutive, we could omit the flushing?

Reply:
Alright, I'll keep it and do a push.